src/main/java/stringcalculator/StringCalculator.java

```diff
@@ -0,0 +1,37 @@
+package stringcalculator;
+
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class StringCalculator {
+    private final Pattern customSeparatorPattern = Pattern.compile("//(.+?)\\n(.*)");
+    private final List<String> defaultSeparators = List.of(":", ",");
+
+    public StringCalculator() {
+    }
+
+    public Integer calculate(String input) {
+        if (Strings.isEmpty(input)) {
+            return 0;
+        }
+
+        List<String> separators = new ArrayList<>(this.defaultSeparators);
+        SeparatorParser separatorParser = new SeparatorParser(this.customSeparatorPattern);
+
+        if (separatorParser.hasCustomSeparator(input)) {
+            CustomSeparator customSeparator = separatorParser.parseSeparator(input);
+            input = customSeparator.getParsedInput();
+            separators.add(customSeparator.getSeparator());
+        }
+
+        String[] splitInputNumbers = StringNumberParser.parse(input, separators);;
```

Comment:
input 에 대한 파싱과정 전부를 StringNumberParser 가 담당해봐도 좋을것 같다고 생각했어요~!!

```suggestion
        String[] splitInputNumbers = StringNumberParser.parse(input);
```

++) `;` <-- 세미콜론이 두개가 있군요! 🧐